README.md

```diff
@@ -1,89 +1,218 @@
 # Summa Aggregation
 
+Summa Aggregation is focused on optimizing the generation of Merkle sum tree, a task identified as the primary time-consuming process in Summa benchmarks. Our goal is to significantly reduce the time required to generate these tree by leveraging parallelization across multiple machines.
+
+The system features `AggregationMerkleSumTree`, a specialized component designed for the efficient construction of complete tree from smaller, aggregated structures known as `mini-tree`. This approach not only speeds up the tree-building process but also enhances scalability and performance in large-scale data environments.
+
 ## Orchestrator
 
-WIP
+The Orchestrator in the Summa Aggregation serves as the central management component, coordinating the data processing activities. It plays a pivotal role in coordinating the activities of `Executors` and `Workers`, improving of tasks in the generation of Merkle sum tree.
+Key functions of the `Orchestrator` include:
+
+- **Dynamic Executor Spawning**: The Orchestrator dynamically spawns `Executors` in numbers set by the user. Each `Executor` is then connected to a dedicated `Worker` for efficient task execution.
+
+- **Task Management and Distribution**: It oversees the overall task flow, loading tasks and distributing them to `Executors`.
+
+- **Error Management and Pipeline Control**: The Orchestrator handles basic pipeline control and responds to errors by initiating the cancellation of all tasks.
+
+## Executor and Worker
+
```

Comment:
I think it is confusing to compare `Executor` and `Worker` because `Executor` is a struct in the codebase why `Worker` is not.